mutiny-core/src/lib.rs

```diff
@@ -355,6 +532,201 @@ impl<S: MutinyStorage> MutinyWallet<S> {
+                                    log_debug!(
+                                        self.logger,
+                                        "could not make payment through federation: {e}"
+                                    );
+                                    continue;
+                                }
+                                _ => {
+                                    log_warn!(self.logger, "unhandled error: {e}")
+                                }
+                            },
+                        }
+                    }
+                }
+                // If payment fails or invoice amount is None or balance is not sufficient, continue to next federation
+            }
+            // If federation client is not found, continue to next federation
+        }
+
+        // If no federation could pay the invoice, fall back to using node_manager for payment
+        self.node_manager
+            .pay_invoice(None, inv, amt_sats, labels)
+            .await
+    }
+
+    /// Creates a BIP 21 invoice. This creates a new address and a lightning invoice.
+    /// The lightning invoice may return errors related to the LSP. Check the error and
+    /// fallback to `get_new_address` and warn the user that Lightning is not available.
+    ///
+    /// Errors that might be returned include:
+    ///
+    /// - [`MutinyError::LspGenericError`]: This is returned for various reasons, including if a
+    ///   request to the LSP server fails for any reason, or if the server returns
+    ///   a status other than 500 that can't be parsed into a `ProposalResponse`.
+    ///
+    /// - [`MutinyError::LspFundingError`]: Returned if the LSP server returns an error with
+    ///   a status of 500, indicating an "Internal Server Error", and a message
+    ///   stating "Cannot fund new channel at this time". This means that the LSP cannot support
+    ///   a new channel at this time.
+    ///
+    /// - [`MutinyError::LspAmountTooHighError`]: Returned if the LSP server returns an error with
+    ///   a status of 500, indicating an "Internal Server Error", and a message stating "Invoice
+    ///   amount is too high". This means that the LSP cannot support the amount that the user
+    ///   requested. The user should request a smaller amount from the LSP.
+    ///
+    /// - [`MutinyError::LspConnectionError`]: Returned if the LSP server returns an error with
+    ///   a status of 500, indicating an "Internal Server Error", and a message that starts with
+    ///   "Failed to connect to peer". This means that the LSP is not connected to our node.
+    ///
+    /// If the server returns a status of 500 with a different error message,
+    /// a [`MutinyError::LspGenericError`] is returned.
+    pub async fn create_bip21(
+        &self,
+        amount: Option<u64>,
+        labels: Vec<String>,
+    ) -> Result<MutinyBip21RawMaterials, MutinyError> {
+        let invoice = if self.config.safe_mode {
+            None
```

Comment:
we can probably enable fedimint in safe mode? probably a better discussion for the future however

Reply:
Nah, in case something is going wrong with fedimint, it would be good for now to leave that unaccessible for now. Until we test more at least.